### PR TITLE
[jasmine] Introduce major version 3

### PR DESCRIPTION
**Pull request by pe8ter**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: See below
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Jasmine 3.3 introduced the function `withContext`, but the typing was added to the 2.8 typings. This PR introduces the split between versions 2 and 3. I removed the `withContext` function from the version 2 typings and made a few small improvements.

To see that Jasmine added the `withContext` function specifically in version 3.3, contrast its docs with version 3.2:
* https://jasmine.github.io/api/3.3/matchers
* https://jasmine.github.io/api/3.2/matchers

### CODE REVIEW

**Comment by armanio123** [comment]
@borisyankov @theodorejb @davidparsson @gmoothart @lukas-zech-software @Engineer2B @cyungmann could any of you please review this PR?

**Comment by pe8ter** [comment]
I found another Jasmine PR made before mine that will slightly conflict with mine: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30261#pullrequestreview-174412416

That one should be merged first, then I can update this PR if necessary.

[ ] Confirm this PR after other Jasmine PR gets merged.

**Review by lukas-zech-software — APPROVED** [review]
nice cleaning up 👍 

will make future change for >v3.3 easier and keeps <v2.8 from breaking

**Comment by lukas-zech-software** [comment]
@pe8ter I approved this PR and also #30261 

As you mentioned, the other one should be merged first to avoid conflicts.

Nice work :+1:

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Jasmine 2.8.0
+// Type definitions for Jasmine 3.3.0
```
Comment (PranavSenthilnathan) [inline types/jasmine/index.d.ts]: The version should only be MAJOR.MINOR (no revision number). This is a rule that should be caught by the dt-header lint rule, so you should turn it on and fix it.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

**Comment by pe8ter** [comment]
@PranavSenthilnathan @andy-ms Is there any way to tune the CI to check the typings only for the files touched in the PR?

**Comment by andy-ms** [comment]
We intentionally test all dependencies -- we don't test every single package. Would you mind fixing the lint error as part of this PR?

**Comment by pe8ter** [comment]
@andy-ms Will do.

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**